File: 17.Backend Security - Everything You Need to Know/code/typescript/③_security_headers_8.ts
import express from "express";
import helmet from "helmet";

const app = express();

app.use(helmet.frameguard({ action: 'deny' }));
app.use(helmet.noSniff());
app.use(helmet.referrerPolicy({ policy: 'strict-origin-when-cross-origin' }));
app.use(helmet.hsts({ maxAge: 63072000, includeSubDomains: true, preload: true }));
app.use(helmet.contentSecurityPolicy({
    directives: {
        defaultSrc: ["'self'"],
        scriptSrc: ["'self'"],
        objectSrc: ["'none'"]
    }
}));

File: 17.Backend Security - Everything You Need to Know/code/rust/③_security_headers_8.rs
use actix_web::{http::header, App};
use actix_web::middleware::DefaultHeaders;

pub fn apply_security_headers() -> DefaultHeaders {
    DefaultHeaders::new()
        .add((header::X_FRAME_OPTIONS, "DENY"))
        .add((header::X_CONTENT_TYPE_OPTIONS, "nosniff"))
        .add((header::REFERRER_POLICY, "strict-origin-when-cross-origin"))
        .add((header::STRICT_TRANSPORT_SECURITY, "max-age=63072000; includeSubDomains; preload"))
        .add((header::CONTENT_SECURITY_POLICY, "default-src 'self'; script-src 'self'; object-src 'none'"))
}

// Router setup logic inside a block
// App::new().wrap(apply_security_headers())
